src/odoc/bin/main.ml

```diff
@@ -87,17 +93,27 @@ end = struct
       match output with
       | Some file ->
         let output = Fs.File.of_string file in
-        if
-          Fs.File.has_ext ".mld" input &&
-          not (Astring.String.is_prefix ~affix:"page-" (Filename.basename file))
+        if Fs.File.has_ext ".mld" input && not (has_page_prefix output)
         then (
           Printf.eprintf "ERROR: the name of the .odoc file produced from a \
                           .mld must start with 'page-'\n%!";
           exit 1
         );
         output
-      | None -> Fs.File.(set_ext ".odoc" input)
+      | None ->
+        let output =
+          if Fs.File.has_ext ".mld" input && not (has_page_prefix input)
+          then
+            let directory = Fs.File.dirname input in
+            let name = Fs.File.basename input in
+            let name = "page-" ^ Fs.File.to_string name in
+            Fs.File.create ~directory ~name
+          else input
+        in
+        Fs.File.(set_ext ".odoc" output)
     in
+    Fmt.epr "Compiling... input=%s output=%s@."
+      (Fs.File.to_string input) (Fs.File.to_string output);
```

Comment:
Is this a temporary print?

Reply:
It was initially, but then I decided to leave it. Is there a reason why odoc doesn't have any info-level logging?

I can remove it if it feels inadequate.

Reply:
No particular reason, except that we haven't set up proper logging yet. Looking at this print, I wonder if the user always sees it. If so, I imagine it's noisy.

Reply:
Having proper logging would be useful for reporting/debugging, let's add it. I'll remove this message until then.